appveyor.yml

```diff
@@ -0,0 +1,324 @@
+version: 1.0.{build}
+branches:
+  only:
+    - NexeyaSGara-windows-ci
```

Comment:
Actually before I invest some time with appveyor I wanted to discuss the practicality of it.
Probably this is not the place for it. I think it should be brought up to a tango kernel meeting.

I will write some advantage/disadvantage table.

Reply:
@tiagocoutinho If you are looking at AppVeyor alternatives, look at [Azur pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/), I know projects that CI happily with MS infra.

Reply:
Thanks @sdebionne. I will have a look

Reply:
I propse to focus just on windows build on this PR.

Appveyor / Azure devops can come in a future PR.

Reply:
How will you do this build, if not with AppVeyor (or Azure CI)?  We need compilation with a whole lot of different versions of Visual Studio.  And ideally in a repeatable environment, so not on your computer.  AppVeyor is already being used for cppTango.   And this PR already includes a configured `appveyor.yml` file, so not much work required at all.

Reply:
> And ideally in a repeatable environment, so not on your computer

I guess in appveyor they keep windows up to date (.NET SDK, VS patches, etc) so thinking the environment is repeatable is, IMHO, an illusion. Same goes for travis. I remember not long ago you were fighting because travis decided to change the base image, right?


> AppVeyor is already being used for cppTango. And this PR already includes a configured appveyor.yml file, so not much work required at all so we need to either start a boost appveyor  project ourselves to do boost 

Sorry but I don't agree. #276 has been stalled for ages because appveyor is not building anymore and nobody has picked this up.
The boost dependencies are not available any more. Do you have an idea how we could do this? I need some help/advice here.
 
I am not saying we should give up on CI. Just saying that we should limit the scope of this PR so it doesn't drag like #276.

Reply:
Boost dependencies should be available in https://github.com/tango-controls/boost-ci/releases page.
The zip files listed there used to be appveyor artifacts.
I've done the same with the latest cppTango release.

Reply:
In cppTango, we have similar dependencies with zmq for instance.
We simply download the corresponding zip files from the [zmq-windows-ci repo release page](https://github.com/tango-controls/zmq-windows-ci/releases
).
Here is the line on appveyor.yml downloading the zip file:
https://github.com/tango-controls/cppTango/blob/tango-9-lts/appveyor.yml#L193

Reply:
Uau! didn't know this existed.

Thanks for unblocking this @bourtemb. That is the kind of push I needed